front/lib/resources/labs_transcripts_resource.ts

```diff
@@ -0,0 +1,283 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import type { LabsTranscriptsProviderType } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+import type { CreationAttributes } from "sequelize";
+
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { LabsTranscriptsConfigurationModel } from "@app/lib/resources/storage/models/labs_transcripts";
+import { LabsTranscriptsHistoryModel } from "@app/lib/resources/storage/models/labs_transcripts";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface LabsTranscriptsConfigurationResource
+  extends ReadonlyAttributesType<LabsTranscriptsConfigurationModel> {}
+export class LabsTranscriptsConfigurationResource extends BaseResource<LabsTranscriptsConfigurationModel> {
+  static model: ModelStatic<LabsTranscriptsConfigurationModel> =
+    LabsTranscriptsConfigurationModel;
+
+  constructor(blob: Attributes<LabsTranscriptsConfigurationModel>) {
+    super(LabsTranscriptsConfigurationModel, blob);
+  }
+
+  static async makeNew(
+    blob: Omit<
+      CreationAttributes<LabsTranscriptsConfigurationModel>,
+      "isActive"
+    >
+  ): Promise<LabsTranscriptsConfigurationResource> {
+    const { userId, connectionId, provider } = blob;
+    const hasExistingConfiguration =
+      await LabsTranscriptsConfigurationModel.count({
+        where: {
+          userId,
+          connectionId,
+          provider,
+        },
+      });
+    if (hasExistingConfiguration) {
+      throw new Error(
+        `A Solution configuration already exists for user ${userId} with connectionId ${connectionId} and provider ${provider}`
+      );
+    }
+    const configuration = await LabsTranscriptsConfigurationModel.create({
+      userId,
+      connectionId,
+      provider,
+      isActive: false,
+    });
+
+    return new LabsTranscriptsConfigurationResource(configuration.get());
+  }
+
+  static async findByUserIdAndProvider({
+    userId,
+    provider,
+  }: {
+    userId: ModelId;
+    provider: LabsTranscriptsProviderType;
+  }): Promise<LabsTranscriptsConfigurationResource | null> {
+    const configuration = await LabsTranscriptsConfigurationModel.findOne({
+      where: {
+        userId,
+        provider,
+      },
+    });
+
+    return configuration
+      ? new LabsTranscriptsConfigurationResource(configuration.get())
+      : null;
+  }
+
+  async setAgentConfigurationId({
+    agentConfigurationId,
+  }: {
+    agentConfigurationId: string | null;
+  }): Promise<
```

Comment:
The return type is not the right one for the function.

Reply:
It could be as simple as:
https://github.com/dust-tt/dust/blob/a3cf539a13fcb6cc5460a541c7855c9056c775a6/connectors/src/resources/base_resource.ts#L46-L55